docs/extending/template_components.md

```diff
@@ -0,0 +1,161 @@
+# Template components
+
+Working with objects that know how to render themselves as elements on an HTML template is a common pattern seen throughout the Wagtail admin. For example, the admin homepage is a view provided by the central `wagtail.admin` app, but brings together information panels sourced from various other modules of Wagtail, such as images and documents (potentially along with others provided by third-party packages). These panels are passed to the homepage via the [`construct_homepage_panels`](../reference/hooks.html#construct-homepage-panels) hook, and each one is responsible for providing its own HTML rendering. In this way, the module providing the panel has full control over how it appears on the homepage.
```

Comment:
```suggestion
Working with objects that know how to render themselves as HTML using a template is a common pattern seen throughout the Wagtail admin. For example, the admin homepage is a view provided by the central `wagtail.admin` app, but brings together information panels sourced from various other modules of Wagtail, such as images and documents (potentially along with others provided by third-party packages). These panels are passed to the homepage via the [`construct_homepage_panels`](../reference/hooks.html#construct-homepage-panels) hook, and each one is responsible for providing its own HTML rendering. In this way, the module providing the panel has full control over how it appears on the homepage.
```

Reply:
The latter version here isn't strictly true - components don't necessarily use templates for their own rendering. (They might just return `mark_safe("<p>some static HTML</p>")` from the `render_html` method instead.) The message I'm aiming to get across here is that we're often faced with a page template, with a bunch of objects to render into it, and we _aren't_ particularly concerned with where those objects come from or how they arrive at their HTML.